app/controllers/admin/npq/applications/notes_controller.rb

```diff
@@ -4,36 +4,40 @@ module Admin
   module NPQ
     module Applications
       class NotesController < Admin::BaseController
-        skip_after_action :verify_authorized
         skip_after_action :verify_policy_scoped
 
-        def edit
-          @npq_application = NPQApplication.find(params[:id])
-        end
+        before_action :assign_npq_application
+
+        def edit; end
 
         def update
-          npq_application = NPQApplication.find(params[:id])
-          npq_application.assign_attributes(note_params)
+          @npq_application = NPQApplication.find(params[:id])
+          @npq_application.assign_attributes(note_params)
 
-          if npq_application.save
-            name = npq_application.participant_identity.user.full_name
+          if @npq_application.save
+            name = @npq_application.participant_identity.user.full_name
 
             flash[:success] = {
               title: "#{name} updated",
-              content: "#{name} has a new note: '#{npq_application.notes.humanize.downcase}' ",
+              content: "#{name} has a new note, see below for details.",
             }
-            redirect_to admin_npq_applications_edge_case_path(npq_application)
           else
             flash[:alert] = {
               title: "#{name} not updated",
               content: "#{name} failed to update",
             }
-            render(:edit)
           end
+          redirect_to admin_npq_applications_edge_case_path(@npq_application)
```

Comment:
Pushing the user here on a failure would mean that the form contents got wiped out, if we render(:edit) instead then the user wouldn't lose everything they've written out if the record fails to save, which might be really important information!

Reply:
thats a fair comment.